tests/unittests/serving/test_experiments_endpoint.py

```diff
@@ -0,0 +1,209 @@
+# -*- coding: utf-8 -*-
+"""Perform functional tests for the REST endpoint `/experiments`"""
+import datetime
+
+from orion.core.worker.trial import Trial
+from orion.storage.base import get_storage
+
+current_id = 0
+
+base_experiment = dict(
+    name='experiment-name',
+    space={'x': 'uniform(0, 200)'},
+    metadata={'user': 'test-user',
+              'orion_version': 'x.y.z',
+              'datetime': datetime.datetime(1, 1, 1),
+              'VCS': {"type": "git",
+                      "is_dirty": False,
+                      "HEAD_sha": "test",
+                      "active_branch": None,
+                      "diff_sha": "diff"}},
+    version=1,
+    pool_size=1,
+    max_trials=10,
+    working_dir='',
+    algorithms={'random': {'seed': 1}},
+    producer={'strategy': 'NoParallelStrategy'},
+)
+
+base_trial = {
+    'experiment': None,
+    'status': 'new',
+    'worker': None,
+    'submit_time': datetime.datetime(1, 1, 1),
+    'start_time': datetime.datetime(1, 1, 1, second=10),
+    'end_time': datetime.datetime(1, 1, 2),
+    'heartbeat': None,
+    'results': [
+        {'name': 'loss', 'type': 'objective', 'value': 0.05},
+        {'name': 'a', 'type': 'statistic', 'value': 10},
+        {'name': 'b', 'type': 'statistic', 'value': 5},
+    ],
+    'params': [
+        {'name': 'x',
+         'type': 'real',
+         'value': 10.0},
+    ],
+}
+
+
+class TestCollection:
+    """Tests the server's response on experiments/"""
+
+    def test_no_experiments(self, client):
+        """Tests that the API returns a positive response when no experiments are present"""
+        response = client.simulate_get('/experiments')
+
+        assert response.json == []
+        assert response.status == "200 OK"
+
+    def test_send_name_and_versions(self, client):
+        """Tests that the API returns all the experiments with their name and version"""
+        expected = [
+            {'name': 'a', 'version': 1},
+            {'name': 'b', 'version': 1}
+        ]
+
+        _add_experiment(name='a', version=1, _id=1)
+        _add_experiment(name='b', version=1, _id=2)
+
+        response = client.simulate_get('/experiments')
+
+        assert response.json == expected
+        assert response.status == "200 OK"
+
+    def test_latest_versions(self, client):
+        """Tests that the API return the latest versions of each experiment"""
+        expected = [
+            {'name': 'a', 'version': 2},
+            {'name': 'b', 'version': 1}
+        ]
+
+        _add_experiment(name='a', version=1, _id=1)
+        _add_experiment(name='a', version=2, _id=2)
```

Comment:
What if version 2 comes before version 1 in the db? Adding 3 versions in order 1, 3, 2 in the database would make sure the response does not simply return the first or last version in order of entry.